### PR TITLE
IAM Role Removal Does Not Require Removal of Permission Boundary

### LINKED ISSUE #959: Deleting an IAM Role Should Not Fail if Permission Boundary Cannot Be Removed
### Summary

I have a role with a permission boundary that disallows removing permission boundaries. This role is then able to create a new role, which inherits that permissions boundary. If I want to delete that role, the `iam_role` plugin fails due to the inability to delete the permission boundary.

Deleting the permission boundary is not actually a prerequisite to deleting the role and there's nothing in AWS that requires removing permission boundaries from roles before deleting them. [This comment](https://github.com/ansible-collections/community.aws/blob/454f5eb5395f1372b125b5b09e731798a698124e/plugins/modules/iam_role.py#L571-L577) is not true when it comes to inherited permission boundaries.

### Issue Type

Bug Report

### Component Name

iam_role

### Ansible Version

```console (paste below)
$ ansible --version
ansible 2.10.8
  config file = None
  configured module search path = [ "elided" ]
  ansible python module location = /usr/local/lib/python3.9/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.9.10 (main, Jan 15 2022, 11:48:00) [Clang 13.0.0 (clang-1300.0.29.3)]
```

### Collection Versions

```console (paste below)
$ ansible-galaxy collection list
amazon.aws                    1.4.1
community.aws                 1.4.0
```

### AWS SDK versions

```console (paste below)
$ pip show boto boto3 botocore
Name: boto3
Version: 1.18.48
Summary: The AWS SDK for Python
Home-page: https://github.com/boto/boto3
Author: Amazon Web Services
Author-email:
License: Apache License 2.0
Location: /usr/lib/python3.7/site-packages
Requires: botocore, jmespath, s3transfer
Required-by:
---
Name: botocore
Version: 1.21.48
Summary: Low-level, data-driven core of boto 3.
Home-page: https://github.com/boto/botocore
Author: Amazon Web Services
Author-email:
License: Apache License 2.0
Location: /usr/lib/python3.7/site-packages
Requires: jmespath, python-dateutil, urllib3
Required-by: awscli, boto3, s3transfer
```

### Configuration

_No response_

### OS / Environment

_No response_

### Steps to Reproduce

```yaml (paste below)
- name: Create Manager Role
  community.aws.iam_role:
    aws_access_key: "{{ aws_access_key }}"
    aws_secret_key: "{{ aws_secret_key }}"
    security_token: "{{ aws_session_token }}"
    name: "manager"
    assume_role_policy_document: |
      {
        "Version": "2012-10-17",
        "Statement": [
          {
            "Effect": "Allow",
            "Principal": {
              "AWS": "{{ aws_account_id }}"
            },
            "Action": "sts:AssumeRole"
          }
        ]
      }
    boundary: |
      {
        "Version": "2012-10-17",
        "Statement": [
          {
            "Sid": "NoBoundaryPolicyDelete",
            "Effect": "Deny",
            "Action": [
                "iam:DeleteRolePermissionsBoundary"
            ],
            "Resource": "*"
          },
        ]
      }
    create_instance_profile: false
  register: manager_role

- name: Assume role
  community.aws.sts_assume_role:
    aws_access_key: "{{ aws_access_key }}"
    aws_secret_key: "{{ aws_secret_key }}"
    aws_security_token: "{{ aws_session_token }}"
    role_arn: "manager"
    role_session_name: "manager"
  register: _ansible_role

- name: Update credentials
  set_fact:
    aws_access_key: "{{ _ansible_role.sts_creds.access_key }}"
    aws_secret_key: "{{ _ansible_role.sts_creds.secret_key }}"
    aws_session_token: "{{ _ansible_role.sts_creds.session_token }}"

- name: Create test role # Inherits permission boundary from manager
  community.aws.iam_role:
    aws_access_key: "{{ aws_access_key }}"
    aws_secret_key: "{{ aws_secret_key }}"
    security_token: "{{ aws_session_token }}"
    name: "test"
    assume_role_policy_document: |
      {
        "Version": "2012-10-17",
        "Statement": [
          {
            "Effect": "Allow",
            "Principal": {
              "AWS": "{{ manager_role.iam_role.arn }}"
            },
            "Action": "sts:AssumeRole"
          }
        ]
      }

- name: Delete test role # Fails because it cannot delete the inherited permission boundary
  community.aws.iam_role:
    aws_access_key: "{{ aws_access_key }}"
    aws_secret_key: "{{ aws_secret_key }}"
    security_token: "{{ aws_session_token }}"
    name: "test"
    state: absent

```

### Expected Results

I expect the be able to delete the role.

### Actual Results

```console (paste below)
Unable to remove permission boundary for role test: An error occurred (AccessDenied) when calling the DeleteRolePermissionsBoundary operation: User: arn:aws:iam::<elided>:role/manager is not authorized to perform: iam:DeleteRolePermissionsBoundary on resource: role test with an explicit deny
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by phene**
##### SUMMARY

Removes unnecessary removal of permission boundary from a role when deleting a role. Unlike inline policies, permission boundaries do not need to be removed from an IAM role before deleting the IAM role. This behavior causes issues when a permission boundary is inherited that prevents removal of the permission boundary.

Fixes #959

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME
iam_role

### CODE REVIEW

**Comment by markuman** [comment]
@phene Thanks for your PR. Can you also please add a bugfixes changelog fragment?  
https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment  
Otherwise LGTM

**Review by markuman — APPROVED** [review]

File: changelogs/fragments/961-iam-role-should-not-remove-permission-boundary-before-deletion.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - Removes unnecessary removal of permission boundary from a role when deleting a role. Unlike inline policies, permission boundaries do not need to be removed from an IAM role before deleting the IAM role. This behavior causes issues when a permission boundary is inherited that prevents removal of the permission boundary. (https://github.com/ansible-collections/community.aws/pull/961)
```
Comment (tremble) [inline changelogs/fragments/961-iam-role-should-not-remove-permission-boundary-before-deletion.yml:2]:
```suggestion
  - iam_role - Removes unnecessary removal of permission boundary from a role when deleting a role. Unlike inline policies, permission boundaries do not need to be removed from an IAM role before deleting the IAM role. This behavior causes issues when a permission boundary is inherited that prevents removal of the permission boundary. (https://github.com/ansible-collections/community.aws/pull/961)
```

**Review by tremble — APPROVED** [review]

**Comment by tremble** [comment]
I've run the integration tests 'locally', and everything still works.

@phene Thanks for the PR, sorry it's taken a while to get it merged.

**Comment by markuman** [comment]
regate

**Pull request merged**